src/Init/Data/Order/Lemmas.lean

```diff
@@ -22,10 +22,41 @@ section AxiomaticInstances
 public instance (r : α → α → Prop) [Asymm r] : Irrefl r where
   irrefl a h := Asymm.asymm a a h h
 
-public instance {r : α → α → Prop} [Total r] : Refl r where
+public instance (r : α → α → Prop) [Total r] : Refl r where
   refl a := by simpa using Total.total a a
 
-public instance Total.asymm_of_total_not {r : α → α → Prop} [i : Total (¬ r · ·)] : Asymm r where
+public instance (r : α → α → Prop) [Asymm r] : Antisymm r where
+  antisymm a b h h' := (Asymm.asymm a b h h').elim
+
+public instance (r : α → α → Prop) [Total r] : Tricho r where
+  tricho a b h h' := by simpa [h, h'] using Total.total (r := r) a b
+
+public theorem Tricho.rel_or_eq_or_swap {r : α → α → Prop} [Tricho r] {a b} :
+    r a b ∨ a = b ∨ r b a := by
+  cases Classical.em (r a b) with | inl hab | inr hab
+  · exact Or.inl hab
+  · cases Classical.em (r b a) with | inl hba | inr hba
+    · exact Or.inr (Or.inr hba)
+    · exact Or.inr (Or.inl (Tricho.tricho _ _ hab hba))
+
+public theorem tricho_of_rel_or_eq_or_swap {r : α → α → Prop}
+    (h : ∀ {a b}, r a b ∨ a = b ∨ r b a) : Tricho r where
+  tricho _ _ hab hba := (h.resolve_left hab).resolve_right hba
+
+public theorem Total.of_not_swap {r : α → α → Prop} [Total r] {a b} (h : ¬ r a b) : r b a :=
+  (Total.total a b).elim (fun h' => (h h').elim) (·)
+
+public def total_of_of_not_swap {r : α → α → Prop} (h : ∀ {a b}, ¬ r a b → r b a) : Total r where
+  total a b := Classical.byCases (p := r a b) Or.inl (fun hab => Or.inr (h hab))
```

Comment:
`Total.of_not_swap` sounds like it returns `Total`. I would rather swap and change these names:

```lean
public theorem rel_of_not_rel_swap {r : α → α → Prop} [Total r] {a b} (h : ¬ r a b) : r b a :=
  (Total.total a b).elim (fun h' => (h h').elim) (·)

public def Total.of_not_rel_swap_imp_rel {r : α → α → Prop} (h : ∀ {a b}, ¬ r a b → r b a) : Total r where
  total a b := Classical.byCases (p := r a b) Or.inl (fun hab => Or.inr (h hab))
```